### PR TITLE
process-compose: init at 0.24.1

### PR DESCRIPTION
**Pull request by thenonameguy**
###### Description of changes

https://github.com/F1bonacc1/process-compose
needed for https://github.com/cachix/devenv/issues/78

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/process-compose/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ config, lib, pkgs, fetchFromGitHub, installShellFiles, ... }:
```
Comment (muscaln) [inline pkgs/applications/misc/process-compose/default.nix:1]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
, installShellFiles
}:
```

File: pkgs/applications/misc/process-compose/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+pkgs.buildGoModule rec {
```
Comment (muscaln) [inline pkgs/applications/misc/process-compose/default.nix:3]:
```suggestion
buildGoModule rec {
```

File: pkgs/applications/misc/process-compose/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    installShellCompletion --cmd process-compose \
+      --bash <($out/bin/process-compose completion bash) \
+      --zsh <($out/bin/process-compose completion zsh) \
+      --fish <($out/bin/process-compose completion fish)
+  '';
+
+  meta = with lib; {
+    description =
+      "Process Compose is like docker-compose, but for orchestrating a suite of processes, not containers.";
```
Comment (muscaln) [inline pkgs/applications/misc/process-compose/default.nix:33]:
```suggestion
    description = "A simple and flexible scheduler and orchestrator to manage non-containerized applications";
```

https://nixos.org/manual/nixpkgs/stable/#var-meta-description

**Review by muscaln — CHANGES_REQUESTED** [review]

**Review by muscaln — APPROVED** [review]
cc @SuperSandro2000 for review.

File: pkgs/applications/misc/process-compose/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  version = "0.24.1";
+
+  src = fetchFromGitHub {
+    owner = "F1bonacc1";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "TKLLq6I+Mcvdz51m8nydTWcslBcQlJCJFoJ10SgfVWU=";
+  };
+
+  ldflags = [ "-X main.version=v${version}" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/process-compose/default.nix:18]: Missing -s -w

File: pkgs/applications/misc/process-compose/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      --zsh <($out/bin/process-compose completion zsh) \
+      --fish <($out/bin/process-compose completion fish)
+  '';
+
+  meta = with lib; {
+    description = "A simple and flexible scheduler and orchestrator to manage non-containerized applications";
+    homepage = "https://github.com/F1bonacc1/process-compose";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ thenonameguy ];
+    mainProgram = "process-compose";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/process-compose/default.nix:40]: mainProgram defaults to pname and should not be repeated

Comment (thenonameguy) [reply pkgs/applications/misc/process-compose/default.nix:40]: addressed in #202314

**Pull request merged**